### PR TITLE
evm: natspec changes and additions

### PR DESCRIPTION
**Pull request by RahulMaganti47**

### CODE REVIEW

**Review by gator-boi — APPROVED** [review]
Lgtm. A little nit picky, but it would be nice if we followed the same indentation rules for multi-line comments. Whether that's indenting (or not) it should be uniform.

**Comment by RahulMaganti47** [comment]
> Lgtm. A little nit picky, but it would be nice if we followed the same indentation rules for multi-line comments. Whether that's indenting (or not) it should be uniform.

yea i prefer the indenting. if we're agreed on that, i can modify the other natspec comments as well

File: evm/src/interfaces/INttManager.sol [hunk 1]
```diff
@@ -43,12 +49,41 @@ interface INttManager is INttManagerState {
...
+    /// @dev Selector 0x2113894.
+    /// @param nttManagerMessageHash The hash of the message.
+    error TrasceiverAlreadyAttestedToMessage(bytes32 nttManagerMessageHash);
+
+    /// @notice Error when the message is not approved.
+    /// @dev Selector 0x451c4fb0.
+    /// @param msgHash The hash of the message.
     error MessageNotApproved(bytes32 msgHash);
+
+    /// @notice Error if the target chain is not supported or does not exist.
```
Comment (djb15) [inline evm/src/interfaces/INttManager.sol:68]: This isn't true actually. The error is emitted when trying to execute a message on an unintended chain: https://github.com/wormhole-foundation/example-native-token-transfers/blob/73c42314aeb1ba58df7f74f9134485c5778ffe2f/evm/src/NttManager/NttManager.sol#L137-L140

File: evm/src/libraries/RateLimiter.sol [hunk 2]
```diff
@@ -9,10 +9,8 @@ import "../libraries/TrimmedAmount.sol";
 
 abstract contract RateLimiter is IRateLimiter, IRateLimiterEvents {
     using TrimmedAmountLib for TrimmedAmount;
-    /**
-     * @dev The duration it takes for the limits to fully replenish
-     */
 
+    /// @dev The duration it takes for the limits to fully replenish.
```
Comment (djb15) [inline evm/src/libraries/RateLimiter.sol:13]: Worth adding "in seconds"?

**Review by djb15 — APPROVED** [review]

File: evm/src/NttManager/TransceiverRegistry.sol [hunk 3]
```diff
@@ -1,7 +1,14 @@
 // SPDX-License-Identifier: Apache 2
 pragma solidity >=0.8.8 <0.9.0;
 
-/// @dev This contract is responsible for handling the registration of Transceivers.
+/// @title TransceiverRegistry
+/// @author Wormhole Project Contributors.
+/// @notice This contract is responsible for handling the registration of Transceivers.
+/// edev This contract a few critical invariants hold when transceivers are added or removed,
```
Comment (gator-boi) [inline evm/src/NttManager/TransceiverRegistry.sol:7]: Tag mistake, missing `@`

Comment (gator-boi) [reply evm/src/NttManager/TransceiverRegistry.sol:7]: Also, this sentence doesn't make sense.

File: evm/src/NttManager/TransceiverRegistry.sol [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+///      including:
+///         1. if a transceiver is not registered, it should be enabled.
```
Comment (gator-boi) [inline evm/src/NttManager/TransceiverRegistry.sol:9]: Capitalize `if` to be consistent.

File: evm/src/Transceiver/WormholeTransceiver/WormholeTransceiver.sol [hunk 5]
```diff
@@ -14,6 +14,18 @@ import "../../interfaces/INttManager.sol";
 
 import "./WormholeTransceiverState.sol";
 
+/// @title WormholeTransceiver
+/// @author Wormhole Project Contributors.
+/// @notice Transceiver implementation for the Wormhole.
```
Comment (gator-boi) [inline evm/src/Transceiver/WormholeTransceiver/WormholeTransceiver.sol:19]: Remove "the"?

File: evm/src/interfaces/INttManager.sol [hunk 6]
```diff
@@ -34,27 +40,56 @@ interface INttManager is INttManagerState {
...
     /// @param mode The mode.
     error InvalidMode(uint8 mode);
 
+    /// @notice Error when the refund to the sender fails.
+    /// @dev Selector 0x2ca23714.
+    /// @param refundAmount The refund amount.
     error RefundFailed(uint256 refundAmount);
+
+    /// @notice Error when the tranceiver already attested to the message.
+    ///         To ensure the client does not continue to initiate calls to the ateestationReceived function.
```
Comment (gator-boi) [inline evm/src/interfaces/INttManager.sol:58]: `attestationReceived` is misspelled.

File: evm/src/interfaces/INttManager.sol [hunk 7]
```diff
@@ -34,27 +40,56 @@ interface INttManager is INttManagerState {
...
+    /// @notice Error when the transfer amount is zero.
+    /// @dev Selector 0x9993626a.
     error ZeroAmount();
+
+    /// @notice Error when the recipient is invalid.
+    /// @dev Selector 0x9c8d2cd2.
     error InvalidRecipient();
+
+    /// @notice Error when the amount burned is different than the balance difference,
+    ///          since NTT does not support burn fees.
```
Comment (gator-boi) [inline evm/src/interfaces/INttManager.sol:83]: Indentation is off.

File: evm/src/interfaces/IWormholeTransceiver.sol [hunk 8]
```diff
@@ -6,29 +6,64 @@ import "../libraries/TransceiverStructs.sol";
...
+
+    /// @notice Error when the VAA has already been consumed.
+    /// @dev Selector: 0x406e719e.
+    /// @param vaaHash The hash of the VAA.
     error TransferAlreadyCompleted(bytes32 vaaHash);
 
-    /// @notice Receive an attested message from the verification layer. This function should verify
-    /// the `encodedVm` and then deliver the attestation to the transceiver NttManager contract.
+    /// @notice Receive an attested message from the verification layer.
+    /// This function should verify the `encodedVm` and then deliver the attestation
```
Comment (gator-boi) [inline evm/src/interfaces/IWormholeTransceiver.sol:60]: Incorrect indentation

File: evm/src/interfaces/IWormholeTransceiver.sol [hunk 9]
```diff
@@ -6,29 +6,64 @@ import "../libraries/TransceiverStructs.sol";
...
+    /// to the transceiver NttManager contract.
     /// @param encodedMessage The attested message.
     function receiveMessage(bytes memory encodedMessage) external;
 
-    /// @notice Parses the encoded instruction and returns the instruction struct. This instruction
-    /// is specific to the WormholeTransceiver contract.
+    /// @notice Parses the encoded instruction and returns the instruction struct.
+    /// This instruction is specific to the WormholeTransceiver contract.
```
Comment (gator-boi) [inline evm/src/interfaces/IWormholeTransceiver.sol:66]: Same here

**Review by gator-boi — APPROVED** [review]

**Review by djb15 — DISMISSED** [review]

**Review by kcsongor — APPROVED** [review]

**Review by djb15 — APPROVED** [review]

**Pull request merged**